### PR TITLE
`bun --inspect`

### PR DESCRIPTION
**Pull request by Jarred-Sumner**
### What does this PR do?

This adds:
- Debugger support via WebKit's Inspector Protocol
- AsyncStackTrace tracking when `--inspect` is enabled, which is manually propagated through the debugger
- Makes various functions used internally appear in profiling when inspected
- Updates WebKit with a new `USE(BUN_JSC_ADDITIONS)` which we can more easily wrap our changes into. In preparation for eventually using the upstream version of WebKit rather than a fork.

Hopefully didn't mess up the merge conflicts but we shall see

### CODE REVIEW

File: src/bun.js/event_loop.zig [hunk 1]
```diff
@@ -487,6 +498,17 @@ pub const GarbageCollectionController = struct {
     };
 };
 
+export fn Bun__tickWhilePaused(paused: *bool) void {
```
Comment (Jarred-Sumner) [inline src/bun.js/event_loop.zig:501]: TODO: remove this

File: src/bun.js/event_loop.zig [hunk 2]
```diff
@@ -487,6 +498,17 @@ pub const GarbageCollectionController = struct {
...
+    JSC.markBinding(@src());
+    JSC.VirtualMachine.get().eventLoop().tickWhilePaused(paused);
+}
+
+comptime {
```
Comment (Jarred-Sumner) [inline src/bun.js/event_loop.zig:506]: TODO: remove this

File: src/bun.js/bindings/webcore/EventEmitter.cpp [hunk 3]
```diff
@@ -234,7 +234,7 @@ void EventEmitter::innerInvokeEventListeners(const Identifier& eventType, Simple
             continue;
 
         WTF::NakedPtr<JSC::Exception> exceptionPtr;
-        JSC::call(lexicalGlobalObject, jsFunction, callData, thisValue, arguments, exceptionPtr);
+        call(lexicalGlobalObject, jsFunction, callData, thisValue, arguments, exceptionPtr);
```
Comment (Jarred-Sumner) [inline src/bun.js/bindings/webcore/EventEmitter.cpp:237]: @paperdave was it intentional we weren't doing async context tracking in EventEmitter?

File: src/bun.js/bindings/napi.cpp [hunk 4]
```diff
@@ -1918,7 +1918,7 @@ extern "C" napi_status napi_call_function(napi_env env, napi_value recv_napi,
     if (thisValue.isEmpty()) {
         thisValue = JSC::jsUndefined();
     }
-    JSC::JSValue result = JSC::call(globalObject, funcValue, callData, thisValue, args);
+    JSC::JSValue result = call(globalObject, funcValue, callData, thisValue, args);
```
Comment (Jarred-Sumner) [inline src/bun.js/bindings/napi.cpp:1921]: @paperdave was it intentional we weren't doing async context tracking in napi?

File: src/js/internal/debugger.ts [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+      var { hostname, port, pathname } = new URL(url);
+      this.url = pathname.toLowerCase();
+    } catch (e) {
+      console.error("[Inspector]", "Failed to parse url", '"' + url + '"');
+      process.exit(1);
+    }
+
+    let portNumber = Number(port);
+    for (let tries = 0; tries < 10; tries++) {
+      const server = Bun.serve<DebuggerWithMessageQueue>({
```
Comment (Jarred-Sumner) [inline src/js/internal/debugger.ts]: TODO: wrap in try/catch

**Comment by Jordan-Hall** [comment]
Will this work with --bun. Currently the debugger not working when forcing node to bun?

File: src/bun.js/bindings/ZigGlobalObject.cpp [hunk 6]
```diff
@@ -3863,7 +3863,7 @@ EncodedJSValue GlobalObject::assignToStream(JSValue stream, JSValue controller)
     arguments.append(stream);
     arguments.append(controller);
 
-    auto result = JSC::call(this, function, callData, JSC::jsUndefined(), arguments);
+    auto result = call(this, function, callData, JSC::jsUndefined(), arguments);
```
Comment (dylan-conway) [inline src/bun.js/bindings/ZigGlobalObject.cpp:3866]: was this meant to be profiledCall?

File: src/bun.js/bindings/bindings.cpp [hunk 7]
```diff
@@ -1811,7 +1812,7 @@ JSC__JSValue JSObjectCallAsFunctionReturnValueHoldingAPILock(JSContextRef ctx, J
         return JSC::JSValue::encode(JSC::JSValue());
 
     NakedPtr<JSC::Exception> returnedException = nullptr;
-    auto result = JSC::call(globalObject, jsObject, callData, jsThisObject, argList, returnedException);
+    auto result = call(globalObject, jsObject, callData, jsThisObject, argList, returnedException);
```
Comment (dylan-conway) [inline src/bun.js/bindings/bindings.cpp:1815]: same here

File: src/bun.js/bindings/ZigGlobalObject.cpp (hunk 6 above)
Comment (Jarred-Sumner) [reply src/bun.js/bindings/ZigGlobalObject.cpp:3866]: no, it calls AsyncContext

**Pull request merged**